src/WorldSelector.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React from 'react';
+import type { WorldName } from './types';
+import type { IntlShape } from 'react-intl';
+import { injectIntl } from 'react-intl';
+
+import {ReactComponent as HeaderIcon} from './svg/SelectWorld.svg'
+import {ReactComponent as RabbitIcon} from './svg/Rabbit.svg';
+import {ReactComponent as RobotIcon} from './svg/Robot.svg';
+import { ReactComponent as SpaceShipIcon } from './svg/SpaceShip.svg';
+
+import './WorldSelector.scss';
+
+type WorldSelectorProps = {
+    intl: IntlShape,
+    onSelect: (value: WorldName) => void
+};
+
+type WorldSelectorState = {
+    world: WorldName
+}
+
+class WorldSelector extends React.Component<WorldSelectorProps, WorldSelectorState> {
+    constructor (props) {
```

Comment:
In this implementation, state for world is used to let App know that there was changes in world, so that world state in App gets updated. However, if we just update world state in App directly from onClick/keyDown events from WorldSelector, logics can be simplified and we don't need to store this state in two separate components. Also, world state is needed for other components as well, so keeping this state in one place, at the parent class (App) would be a nice refactoring.

Reply:
I suggest removing WorldSelectorState type, constructor, as well as componentDidUpdate method and use WorldSelector as a UI component.

Reply:
Makes sense, and done.